backend/backend/gmaps.py

```
@@ -61,19 +157,19 @@ def get_distance_matrix_api_payload(
     origins: list[Location],
     destinations: list[Location],
     travel_mode: TravelMode = TravelMode.DRIVE,
+    departure_time: str | None = None,
+
 ):
     payload = {
         "origins": [l.to_route_matrix_location() for l in origins],
         "destinations": [l.to_route_matrix_location() for l in destinations],
         "travelMode": str(travel_mode),
-        # TODO: set departureTime for travel_mode=TRANSIT. Otherwise, it defaults to
-        # now, which is not reproducible. But you can only specify datetimes close
-        # to the current moment, which makes things more complicated. It'd be ideal to
-        # use something like "Monday 10AM", but then you also need to take into account
-        # the timezone and find the closest Monday.
-        # https://developers.google.com/maps/documentation/routes/transit-route#options
     }
 
+    # If travel_mode is TRANSIT, we can specify a departure time.
+    if travel_mode == TravelMode.TRANSIT and departure_time is not None:
+        payload["departureTime"] = departure_time
```

Let's raise an error if the departure time is set but the travel mode is not `TRANSIT`. It suggests there's been an error, and errors should never pass silently :)